Documentation/coding-guidelines/adding-api-guidelines.md

```diff
@@ -0,0 +1,124 @@
+  - If your package depends directly on runtime changes or library changes that ship with the runtime
+    (i.e. System.Private.CoreLib) then your target framework should be `netstandard<latest>` (hint: usually indicated by needing the latest Microsoft.TargetingPack.Private.* package)
+  - If your package depends on changes in a native shim package then your target framework should be
+    `netstandard<latest>`.
+  - When targeting `netstandardX` your new API must be supported by all target frameworks that
+    map to that netstandard version (see [mapping table](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md#mapping-the-net-platform-standard-to-platforms)). If not bump the version to the minimum netstandard version that supports this API on all frameworks that map to that netstandard version.
+- If the library is not part of netstandard and not building against netstandard
+  - All the rules for a library that is not part of netstandard apply but instead of having a target framework of
+    `netstandard<latest>` it will have a target framework of `<framework><latest>`. Example `net<latest>` for desktop or `netcoreapp<latest>` for .NET Core.
+- It is common for a library to target multipe frameworks all the latest versions when adding new APIs.
+
+###Determine library version
+- If targeting netstandard
+  - Ensure minor version of the library is bumped since last stable package release
+- If targeting netcoreapp
+  - No version bump necessary
+
+##Making the changes in repo
+
+**If changing the library version**
+  - Update the `AssemblyVersion` property in `<Library>\dir.props` (ex: [System.Runtime\dir.props](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/dir.props#L4)). If `AssemblyVersion` is
+    set in any of the individual csproj's remove it from them.
+
+**Update ref**
+  - If changing target framework
+    - `NugetTargetMoniker` in `<Library>\ref\<Library>.csproj` (ex: [ref\System.Runtime.csproj](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/System.Runtime.csproj#L8))
+    - Update framework section in `<Library>\ref\project.json` (ex: [System.Runtime\ref\project.json](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/project.json#L4))
+  - Add your API to `<Library>\ref\<Library>.cs`. Use GenAPI to normalize the file.
+  - If adding an additional target framework for the ref (ex: adding a .NET Core specific API) you may need
+  to add a `<Library>\ref\<Library>.builds` for the new configuration (ex: [ref/System.Runtime.Extensions.builds](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/ref/System.Runtime.Extensions.builds#L6)).
+
+**Update src**
+  - If changing target framework
+    - Update `NugetTargetMoniker` in `<Library>\src\<Library>.csproj` (ex: [src\System.Runtime.csproj](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/System.Runtime.csproj#L12))
```

Comment:
Conditions on TargetGroup are only needed for projects that cross-compile.  In the case a project is cross-compiling it will want to use the NuGetTargetMoniker that is derived from the TargetGroup.